File: backend/app/schemas/salary_usage.py
"""
薪资使用记录 Schemas
"""
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional


class SalaryUsageBase(BaseModel):
    """薪资使用记录基础 Schema"""
    usage_type: str = Field(..., description="使用类型: housing/food/transport/shopping/entertainment/medical/education/other")
    amount: float = Field(..., gt=0, description="使用金额（明文，前端传入）")
    usage_date: datetime = Field(..., description="使用日期")
    description: Optional[str] = Field(None, max_length=500, description="备注说明")


class SalaryUsageCreate(SalaryUsageBase):
    """创建薪资使用记录"""
    salary_record_id: str = Field(..., description="薪资记录ID")


class SalaryUsageUpdate(BaseModel):
    """更新薪资使用记录"""
    usage_type: Optional[str] = Field(None, description="使用类型")
    amount: Optional[float] = Field(None, gt=0, description="使用金额")
    usage_date: Optional[datetime] = Field(None, description="使用日期")
    description: Optional[str] = Field(None, max_length=500, description="备注说明")


class SalaryUsageInDB(SalaryUsageBase):
    """数据库中的薪资使用记录（包含ID和时间戳）"""
    id: str
    user_id: str
    salary_record_id: str
    amount: str  # 加密后的金额
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SalaryUsageResponse(SalaryUsageBase):
    """薪资使用记录响应（返回给前端）"""
    id: str
    salary_record_id: str
    amount: float  # 解密后的金额
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SalaryUsageListResponse(BaseModel):
    """薪资使用记录列表响应"""
    total: int = Field(..., description="总记录数")
    items: list[SalaryUsageResponse] = Field(..., description="记录列表")

File: miniapp/src/api/salary-usage.ts
/**
 * 薪资使用记录 API
 * 与 backend /api/v1/salary-usage 一致
 */
import request from '@/utils/request'

const PREFIX = '/api/v1/salary-usage'

/** 消费类型 */
export type UsageType =
  | 'housing'
  | 'food'
  | 'transport'
  | 'shopping'
  | 'entertainment'
  | 'medical'
  | 'education'
  | 'other'

export interface SalaryUsageRecord {
  id: string
  user_id: string
  salary_record_id: string
  usage_type: UsageType
  amount: number
  usage_date: string
  description: string | null
  created_at: string
  updated_at: string
}

export interface SalaryUsageCreate {
  salary_record_id: string
  usage_type: UsageType
  amount: number
  usage_date: string
  description?: string | null
}

export interface SalaryUsageUpdate {
  usage_type?: UsageType
  amount?: number
  usage_date?: string
  description?: string | null
}

export interface SalaryUsageListParams {
  salary_record_id?: string
  usage_type?: UsageType
  from_date?: string
  to_date?: string
  limit?: number
  offset?: number
}

/** 列表 */
export function listSalaryUsage(params?: SalaryUsageListParams) {
  let url = PREFIX
  if (params && Object.keys(params).length) {
    const queryParts: string[] = []
    if (params.salary_record_id)
      queryParts.push(`salary_record_id=${encodeURIComponent(params.salary_record_id)}`)
    if (params.usage_type) queryParts.push(`usage_type=${encodeURIComponent(params.usage_type)}`)
    if (params.from_date) queryParts.push(`from_date=${encodeURIComponent(params.from_date)}`)
    if (params.to_date) queryParts.push(`to_date=${encodeURIComponent(params.to_date)}`)
    if (params.limit != null) queryParts.push(`limit=${encodeURIComponent(String(params.limit))}`)
    if (params.offset != null)
      queryParts.push(`offset=${encodeURIComponent(String(params.offset))}`)
    const qs = queryParts.join('&')
    if (qs) url += (url.includes('?') ? '&' : '?') + qs
  }
  return request<SalaryUsageRecord[]>({ url, method: 'GET' })
}

/** 新增 */
export function createSalaryUsage(data: SalaryUsageCreate) {
  return request<SalaryUsageRecord>({ url: PREFIX, method: 'POST', data })
}

/** 单条 */
export function getSalaryUsage(recordId: string) {
  return request<SalaryUsageRecord>({ url: `${PREFIX}/${recordId}`, method: 'GET' })
}

/** 更新 */
export function updateSalaryUsage(recordId: string, data: SalaryUsageUpdate) {
  return request<SalaryUsageRecord>({ url: `${PREFIX}/${recordId}`, method: 'PUT', data })
}

/** 删除 */
export function deleteSalaryUsage(recordId: string) {
  return request<void>({ url: `${PREFIX}/${recordId}`, method: 'DELETE' })
}
